cognite/neat/_session/_issues.py

```diff
@@ -0,0 +1,464 @@
+import json
+from collections import defaultdict
+from typing import Any
+
+from cognite.neat._issues import IssueList
+from cognite.neat._store import NeatStore
+
+
+class Issues:
+    """Class to handle issues in the NeatSession."""
+
+    def __init__(self, store: NeatStore) -> None:
+        self._store = store
+
+    @property
+    def _issues(self) -> IssueList:
+        """Get all issues from the last change in the store."""
+        issues = IssueList()
+        if change := self._store.provenance.last_change:
+            issues += change.errors or IssueList()
+            issues += change.issues or IssueList()
+        return issues
+
+    @property
+    def _stats(self) -> dict[str, Any]:
+        """Compute statistics about issues."""
+        by_type: defaultdict[str, int] = defaultdict(int)
+        by_code: defaultdict[str, int] = defaultdict(int)
+
+        stats: dict[str, Any] = {
+            "total": len(self._issues),
+            "by_type": by_type,
+            "by_code": by_code,
+            "severity_order": ["ModelSyntaxError", "ConsistencyError", "Recommendation"],
+        }
+
+        for issue in self._issues:
+            issue_type = type(issue).__name__
+            stats["by_type"][issue_type] += 1
+
+            if issue.code:
+                stats["by_code"][f"{issue_type}:{issue.code}"] += 1
+
+        return stats
+
+    @property
+    def _serialized_issues(self) -> list[dict[str, Any]]:
+        """Convert issues to JSON-serializable format."""
+        serialized = []
+        for idx, issue in enumerate(self._issues):
+            issue_type = getattr(issue, "type", type(issue).__name__)
```

Comment:
Issue type is defined differently here. Lets use `type(issue).__name__` both places. Or add a classmethod to issue  

```python
@classmethod
def issue_type(cls) -> str
    return cls.__name__
```